cmd/gateway-main.go

```diff
@@ -281,10 +281,9 @@ func StartGateway(ctx *cli.Context, gw Gateway) {
 
 	// Create new notification system.
 	globalNotificationSys = NewNotificationSys(globalServerConfig, globalEndpoints)
-	if globalEtcdClient != nil {
+	if globalEtcdClient != nil && gatewayName == "nas" {
```

Comment:
you can do:
`if globalEtcdClient != nil && newObj.IsNotificationSupported() {...}`